bitcoin/src/util/mod.rs

```diff
@@ -120,3 +119,54 @@ pub mod address {
 
 #[deprecated(since = "0.30.0", note = "Please use crate::bip158")]
 pub use crate::bip158;
+
+/// The `misc` module was moved and re-named to `sign_message`.
+#[deprecated(since = "0.30.0", note = "Please use crate::sign_message")]
+pub mod misc {
+    pub use crate::sign_message::*;
+}
+
+/// Search for `needle` in the vector `haystack` and remove every
+/// instance of it, returning the number of instances removed.
+/// Loops through the vector opcode by opcode, skipping pushed data.
+// For why we deprecated see: https://github.com/rust-bitcoin/rust-bitcoin/pull/1259#discussion_r968613736
+#[deprecated(since = "0.30.0", note = "No longer supported")]
+pub fn script_find_and_remove(haystack: &mut Vec<u8>, needle: &[u8]) -> usize {
```

Comment:
I don't see a point in deprecating an item if it's not accessible from the original place. I'd either re-export it or remove it completely.

Reply:
Oof, I didn't even notice that this wasn't re-exported :P 

We should re-export it.

Reply:
agh, thanks, that was a straight up mistake. I've moved it _inside_ the `misc` module where it should have been. Also I've moved the deprecated to the `sign_message` wildcard instead of the `misc` module.